src/Components/Server/src/ComponentHub.cs

```diff
@@ -242,21 +242,6 @@ public async ValueTask<bool> ReceiveJSDataChunk(long streamId, long chunkId, byt
             return await circuitHost.ReceiveJSDataChunk(streamId, chunkId, chunk, error);
         }
 
-        public async ValueTask DispatchBrowserEvent(JsonElement eventInfo)
```

Comment:
Removing this method gives me a pause - I can't find the history for this change (it ends at this PR - https://github.com/dotnet/aspnetcore/pull/12250), but I vaguely recall we used to plumb events thru JSInterop but special cased it at as a perf optimization. Are we losing out on those benefits as part of this change?

Reply:
The only difference we had before by not using JS interop was more scenario-specific logs, which I don't think are very consequential.

But the more fundamental issue is that is *has* to go via JS interop if we want it to also have JS interop features like transporting `byte[]` or DotNetObjectReference.

Reply:
As for perf, it's not going to be meaningfully different because even before this change, we were JSON-serializing the data before sending it over BlazorPack, which is still equivalent to what JS interop does.

Reply:
Update: I should be clearer about this because there *is* some extra cost in going through JS interop, but (1) it's the same as we had in 5.0, so not a regression - just not an improvement either, and (2) this sets us up to improve it further in the future if we want.

Overview for Blazor Server:

| Version | How event dispatch worked | Summary of transformations |
| --- | --- | --- |
| 5.0 | JS sent JSON strings; .NET received .NET strings then called STJ to build .NET objects | JS JSON string -> Encode as UTF8 for wire -> Decode as .NET string for hub method -> Encode as UTF8 for JSON parsing -> .NET EventData object |
| 6.0-pre7 | JS sent UTF8-encoded bytes; .NET received bytes then called STJ to build .NET objects | JS JSON string -> Encode as UTF8 for wire -> Receive as UTF8 on hub method -> JSON parse to .NET EventData object |
| This PR | JS sends JSON strings; .NET receives .NET strings then our JS interop code serializes as UTF8 and our custom deserialization logic processes the JsonElement over it | JS JSON string -> Encode as UTF8 for wire -> Decode as .NET string for hub method -> JSInterop does own UTF8 encoding and passes that to STJ -> JsonElement -> .NET EventData object |

So in effect, this PR removes an optimization we had previously added in a 6.0 preview and takes us back to the same perf characteristics we had in 5.0 and before, in that it goes back to passing the data via a .NET string representation instead of staying as UTF8 all the way through.

Of course it would be nice if we didn't have to go through that extra representation, but it's fundamental to how JS interop is defined in terms of exchanging strings (JS strings and .NET strings, not UTF8 bytes). But the good news it that following the layering properly like this PR does means we can improve that layering in the future and would automatically get benefits here. We could eliminate the .NET string representation entirely, as suggested at https://github.com/dotnet/aspnetcore/issues/35065.